cnf-tests/pod-utils/hugepages-allocator/main.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"math"
+	"os"
+	"syscall"
+	"time"
+	"unsafe"
+
+	"k8s.io/klog/v2"
+)
+
+const (
+	// MapHugeShift Shift for HugePage size
+	MapHugeShift = 26
+	// DefaultHugePageSize 1GB HugePage size
+	DefaultHugePageSize = 1 * 1024 * 1024 * 1024
+)
+
+// MAP_HUGE_1GB 1GB HugePage mmap flag
+const MAP_HUGE_1GB = 30 << MapHugeShift
+
+type Args struct {
+	Sleep        time.Duration
+	HugePageSize int
+}
+
+func main() {
+	klog.InitFlags(nil)
+	args := &Args{}
+	flag.DurationVar(&args.Sleep, "timeout", math.MaxInt64, "set the sleep in seconds for program to sleep - sleep forever by default")
+	flag.IntVar(&args.HugePageSize, "hugepage-size", DefaultHugePageSize, "hugepage size to allocate - allocate 1G by default")
+
+	flag.Parse()
+
+	// Flags for HugePage allocation
+	mmapFlags := syscall.MAP_PRIVATE | syscall.MAP_ANONYMOUS | syscall.MAP_HUGETLB | MAP_HUGE_1GB
+	// Use mmap to allocate memory
+	addr, _, errno := syscall.Syscall6(
+		syscall.SYS_MMAP,
+		0,                          // Let the kernel choose the address
+		uintptr(args.HugePageSize), // Size of the memory
+		uintptr(syscall.PROT_READ|syscall.PROT_WRITE), // Read/Write permissions
+		uintptr(mmapFlags), // mmap flags
+		0,                  // File descriptor (not used for anonymous memory)
+		0,                  // Offset
+	)
+	if errno != 0 {
+		// there's little we can do here
+		_, _ = fmt.Fprintf(os.Stderr, "Failed to allocate hugepage. errno=%v\n", errno)
```

Comment:
should we print using the log package?

Reply:
I thought it would be a good idea to print to stderr because when we exec the command in the pod we usually expect some data in stderr/sdtout.

How do you suggest to handle this though?

Reply:
The command is in the pod definition we don't do exec into it.
we can just do `logs` and we will see all the logs from the pod. you already used the klog lib in the code so you can just use it here I think :)